export/wiki/spawn_groups/structs.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any, Dict, Iterable, List, Optional, Tuple
+
+from automate.hierarchy_exporter import ExportModel, Field
+from export.wiki.maps.models import WeighedClassSwap
+from ue.asset import UAsset
+from ue.properties import ArrayProperty
+from ue.utils import sanitise_output
+
+__all__ = [
+    'NpcEntry',
+    'NpcGroup',
+    'NpcLimit',
+    'convert_single_class_swap',
+    'convert_class_swaps',
+    'convert_group_entry',
+    'convert_limit_entries',
+]
+
+
+def _zip_swap_outputs(d: Dict[str, Any]) -> Iterable[Tuple[float, Optional[str]]]:
+    npcs: ArrayProperty = d['ToClasses']
+    weights = d['Weights'].values
+
+    for index, kls in enumerate(npcs.values):
+        # Get weight of this class. Defaults to 1 if array is too short.
+        weight = float(weights[index]) if index < len(weights) else 1.0
+        yield (weight, sanitise_output(kls))
```

Comment:
Did you look at Python's `zip_longest` as an alternative here? Your approach may be neater given the extra work needed, but wanted you to be aware that the function exists.